plugin/vcs/github/github.go

```diff
@@ -0,0 +1,286 @@
+}
+
+func (p *Provider) FetchRepositoryFileList(ctx context.Context, oauthCtx common.OauthContext, instanceURL, repositoryID, ref, filePath string) ([]*vcs.RepositoryTreeNode, error) {
+	return nil, errors.New("not implemented yet") // TODO: https://github.com/bytebase/bytebase/issues/928
+}
+
+func (p *Provider) CreateFile(ctx context.Context, oauthCtx common.OauthContext, instanceURL, repositoryID, filePath string, fileCommit vcs.FileCommitCreate) error {
+	return errors.New("not implemented yet") // TODO: https://github.com/bytebase/bytebase/issues/928
+}
+
+func (p *Provider) OverwriteFile(ctx context.Context, oauthCtx common.OauthContext, instanceURL, repositoryID, filePath string, fileCommit vcs.FileCommitCreate) error {
+	return errors.New("not implemented yet") // TODO: https://github.com/bytebase/bytebase/issues/928
+}
+
+func (p *Provider) ReadFile(ctx context.Context, oauthCtx common.OauthContext, instanceURL, repositoryID, filePath, commitID string) (string, error) {
+	return "", errors.New("not implemented yet") // TODO: https://github.com/bytebase/bytebase/issues/928
+}
+
+func (p *Provider) ReadFileMeta(ctx context.Context, oauthCtx common.OauthContext, instanceURL, repositoryID, filePath, branch string) (*vcs.FileMeta, error) {
+	return nil, errors.New("not implemented yet") // TODO: https://github.com/bytebase/bytebase/issues/928
+}
+
+func (p *Provider) CreateWebhook(ctx context.Context, oauthCtx common.OauthContext, instanceURL, repositoryID string, payload []byte) (string, error) {
+	return "", errors.New("not implemented yet") // TODO: https://github.com/bytebase/bytebase/issues/928
+}
+
+func (p *Provider) PatchWebhook(ctx context.Context, oauthCtx common.OauthContext, instanceURL, repositoryID, webhookID string, payload []byte) error {
+	return errors.New("not implemented yet") // TODO: https://github.com/bytebase/bytebase/issues/928
+}
+
+func (p *Provider) DeleteWebhook(ctx context.Context, oauthCtx common.OauthContext, instanceURL, repositoryID, webhookID string) error {
+	return errors.New("not implemented yet") // TODO: https://github.com/bytebase/bytebase/issues/928
+}
+
+func httpGet(ctx context.Context, client *http.Client, resourcePath string, token *string, oauthCtx oauthContext, refresher common.TokenRefresher) (code int, respBody string, err error) {
+	return retry(ctx, client, token, oauthCtx, refresher,
+		func() (*http.Response, error) {
+			url := fmt.Sprintf("%s/%s", apiURL, resourcePath)
+			req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+			if err != nil {
+				return nil, errors.Wrapf(err, "construct GET %s", url)
+			}
+
+			req.Header.Set("Content-Type", "application/json")
+			req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", *token))
+			resp, err := client.Do(req)
+			if err != nil {
+				return nil, errors.Wrapf(err, "GET %s", url)
+			}
+			return resp, nil
+		},
+	)
+}
+
+const maxRetries = 3
```

Comment:
how about putting this and api url to the top constant section, similar to gitlab/gitlab.go

Reply:
I purposely move this down here, as making it closer to the place where it is being used. I even wanted to move inside the `retry` function... since it is being referenced nowhere else. But happy to accommodate.

Reply:
yeah, let's move it to the top to be consistent with GitLab. We might be able to unify the retry policy and code later.